sql/src/main/java/io/crate/executor/transport/TransportShardUpsertAction.java

```diff
@@ -150,7 +149,7 @@ public TransportShardUpsertAction(Settings settings,
                 shardResponse.add(location);
             } catch (Exception e) {
                 if (retryPrimaryException(e)) {
-                    Throwables.propagate(e);
+                    throw new RuntimeException(e);
```

Comment:
I think in this case `e` is already a RuntimeException, so it should be unnecessary to re-wrap it. And I'm not sure if it's even safe to do so.

I'd suggest to keep using `propagate` here for now and look into this separately